utils/data.py

```diff
@@ -31,6 +31,14 @@ async def process_commands(self, message):
 
         ctx = await self.get_context(message)
         await self.invoke(ctx)
+    
+    ## @story{74} Bot will send a message to the first listed text channel 
+    # when a new user joins the server.   
+    async def on_member_join(self, member):
+        server = member.guild
+        channel = server.channels[0].text_channels[0]
```

Comment:
The first listed text channel is not necessarily appropriate. For example, the bot may not have permission to post to that channel, or the channel may be used as a place to display rules (and therefore inappropriate for welcome messages even if the bot does have permissions). A better setup would allow server admins to specify, by way of a command, which channel to send the messages to. Often, you want it to be a public channel where the new member can respond to the message.

However, for our project and time crunch, this is sufficient for us. Just keep these concerns in mind if you ever do a bot of your own in the future.

Reply:
Yeah I tried to find a more optimal solution. Given that there is no "default channel" attribute in the discord.py api for guilds/servers, this seemed to be the best I could do without significant effort or added commands like you mentioned.

Reply:
That's fair. We don't have time for a better solution at this point. I'll work on improving this before I put the bot into use.